kotlin/buffer/src/main/java/com/esri/arcgisruntime/sample/buffer/MainActivity.kt

```diff
@@ -0,0 +1,148 @@
+        }
+
+        // add overlays to the mapView.
+        mapView.graphicsOverlays.apply {
+            add(geodesicGraphicsOverlay)
+            add(planarGraphicsOverlay)
+            add(tapLocationsOverlay)
+        }
+
+        // create a buffer around the clicked location
+
+        mapView.onTouchListener = object: DefaultMapViewOnTouchListener(applicationContext, mapView){
+            override fun onSingleTapConfirmed(motionEvent: MotionEvent): Boolean {
+
+                // get the point that was clicked and convert it to a point in the map
+                val screenPoint = android.graphics.Point(motionEvent.x.roundToInt(),
+                        motionEvent.y.roundToInt())
+                val mapPoint = mapView.screenToLocation(screenPoint)
+
+                // only draw a buffer if a value was entered
+                if (bufferInput.text.toString().isNotEmpty()) {
+                    // get the buffer distance (miles) entered in the text box.
+//                    val bufferInMiles = Double.valueOf(bufferInput.getText().toString())
+                    val bufferInMiles = bufferInput.text.toString().toDouble()
+
+                    // convert the input distance to meters, 1609.34 meters in one mile
+                    val bufferInMeters = bufferInMiles * 1609.34
+
+                    // create a planar buffer graphic around the input location at the specified distance
+                    val bufferGeometryPlanar = GeometryEngine.buffer(mapPoint, bufferInMeters)
+                    val planarBufferGraphic = Graphic(bufferGeometryPlanar)
+
+                    // create a geodesic buffer graphic using the same location and distance
+                    val bufferGeometryGeodesic = GeometryEngine.bufferGeodetic(mapPoint, bufferInMeters,
+                            LinearUnit(LinearUnitId.METERS), Double.NaN, GeodeticCurveType.GEODESIC)
+                    val geodesicBufferGraphic = Graphic(bufferGeometryGeodesic)
+
+                    // create a graphic for the user tap location
+                    val locationGraphic = Graphic(mapPoint)
+
+                    // add the buffer polygons and tap location graphics to the appropriate graphic overlays.
+                    planarGraphicsOverlay.graphics.add(planarBufferGraphic)
+                    geodesicGraphicsOverlay.graphics.add(geodesicBufferGraphic)
+                    tapLocationsOverlay.graphics.add(locationGraphic)
+                } else {
+                    Toast.makeText(this@MainActivity, "Please enter a buffer distance first.", Toast.LENGTH_LONG).show()
+                }
+                return true
+            }
+        }
+
+        clearButton.setOnClickListener {
+            planarGraphicsOverlay.graphics.clear()
+            geodesicGraphicsOverlay.graphics.clear()
+            tapLocationsOverlay.graphics.clear()
+        }
+    }
+
+    override fun onPause() {
+        super.onPause()
+        mapView.pause()
```

Comment:
We just try to order the lifecycle calls like in here: https://devtopia.esri.com/runtime/common-samples/wiki/Android-Samples-Repo%3A-A-Contribution-Guide#handle-life-cycles